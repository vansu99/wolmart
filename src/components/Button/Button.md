### Examples

**Outline - no background**

`<Button typeBtn="submit" outline content="đăng ký" />`


**Width full**

`<Button typeBtn="submit" wFull content="đăng ký" />`


**Size button**

`<Button typeBtn="submit" size="small" content="đăng ký" />`

`<Button typeBtn="submit" size="medium" content="đăng ký" />`

`<Button typeBtn="submit" size="large" content="đăng ký" />`


**Disabled button**

`<Button typeBtn="submit" disabled content="đăng ký" />`


**Type button**

`<Button typeBtn="submit" typeBtn="button" content="đăng ký" />`

`<Button typeBtn="submit" typeBtn="submit" content="đăng ký" />`

`<Button typeBtn="submit" typeBtn="reset" content="đăng ký" />`


**Event click button**

`<Button typeBtn="submit" wFull content="đăng ký" @click="onSubmit" />`
