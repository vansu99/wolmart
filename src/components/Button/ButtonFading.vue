<template>
  <span class="button--fading">{{ content }}</span>
</template>

<script>
export default {
  props: {
    content: String,
  },
};
</script>

<style lang="scss" scoped>
.button--fading {
  display: inline-block;
  padding: 1rem 1.8rem;
  width: 100%;
  border: 0.2rem solid #ccc;
  border-radius: 0.3rem;
  background: transparent;
  font-family: $font-primary;
  font-size: 1.3rem;
  font-weight: 600;
  text-transform: uppercase;
  color: $text-primary;
  line-height: 1;
  transition: all 0.3s ease-in-out;
  cursor: pointer;
  &:hover {
    color: $text-white-light;
    border: 0.2rem solid $bg-second;
    background-color: $bg-second;
  }
}
</style>