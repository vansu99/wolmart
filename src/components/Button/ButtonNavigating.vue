<template>
  <span class="button--opacity"
    ><i v-show="left" class="fas fa-chevron-left left"></i
    ><i v-show="left" class="fas fa-chevron-left left"></i>
    {{ content }}
    <i v-show="!left" class="fas fa-chevron-right right"></i
    ><i v-show="!left" class="fas fa-chevron-right right"></i>
  </span>
</template>

<script>
export default {
  props: { content: String, left: Boolean },
};
</script>

<style lang="scss" scoped>
.button--opacity {
  display: inline-block;
  padding: 1.3rem 2.8rem;
  border: 2px solid $bg-second;
  border-radius: 0.3rem;
  background-color: $bg-second;
  font-family: $font-primary;
  font-size: 1.4rem;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
  color: #fff;
  transition: all 0.3s ease-in-out;
  cursor: pointer;
  &:hover {
    opacity: 0.95;
  }
  & i {
    font-size: 1.2rem;
    transition: transform 0.3s ease-in-out;
    &:nth-child(3) {
      margin-left: 0.3rem;
    }
    &:nth-child(2) {
      margin-right: 0.3rem;
    }
  }
  &:hover .right {
    transform: translateX(1rem);
  }
  &:hover .left {
    transform: translateX(-1rem);
  }
}
</style>