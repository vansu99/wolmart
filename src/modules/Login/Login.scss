.login {
  @include flexCenter;
  height: 100%;
  &-wrapper {
    width: 35rem;
    padding: 3rem 3.5rem;
  }
  &-heading {
    font-size: 2.6rem;
    margin-bottom: 1.5rem;
    color: $hover-color;
    font-weight: bold;
  }
  &-error {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
    border-radius: 4px;
    margin-bottom: 2rem;
    font-size: 1.3rem;
    padding: 10px;
  }
  &-form {
    .form-group {
      margin-bottom: 2rem;
    }
    .form-label {
      display: block;
      font-size: 1.6rem;
      margin-bottom: 1rem;
    }
    .form-input {
      width: 100%;
      padding: 10px 10px 10px 15px;
      font-family: $font-primary;
      color: $text-primary;
      outline: none;
      border: 1px solid #c9c9c9;
      border-radius: 4px;
      overflow: hidden;
      background-color: transparent;
      transition: border-color 0.2s ease-in;
      &:focus {
        border-color: $hover-color;
      }
    }
    .form-error {
      display: block;
      color: $bg-danger;
      font-size: 1.3rem;
      font-weight: 500;
      margin-top: 10px;
    }
    .form-forgot {
      color: $text-primary;
      font-size: 1.3rem;
      margin-bottom: 1.5rem;
      b {
        color: $bg-primary;
        font-weight: bold;
      }
    }
    .form-button {
      width: 100%;
      font-size: 1.5rem;
      font-weight: 500;
      font-family: $font-primary;
      display: block;
      padding: 12px;
      background-color: $bg-primary;
      border: 0;
      outline: 0;
      color: #fff;
      line-height: 1.2;
      cursor: pointer;
      border-radius: 4px;
      text-transform: capitalize;
      &:hover {
        opacity: 0.95;
      }
    }
  }
  .line-text {
    text-align: center;
    position: relative;
    padding: 1rem 0;
    span {
      display: inline-block;
      background-color: white;
      position: relative;
      z-index: 2;
      padding: 1rem;
      font-weight: 600;
      font-size: 1.5rem;
      text-transform: capitalize;
      color: $bg-second;
    }
    &:before {
      content: "";
      width: 100%;
      height: 1px;
      position: absolute;
      top: 50%;
      left: 0;
      transform: translateY(-50%);
      background-color: $bg-second;
    }
  }
  &-social {
    .social-item {
      cursor: pointer;
      @include flexCenter;
      padding: .6rem 0;
      font-size: 1.5rem;
      margin-bottom: 1.2rem;
      border-radius: 4px;
      border: 1px solid $text-light;
      color: $text-primary;
      background-color: #fff;
      svg {
        margin-right: 1rem;
      }
      &:hover {
        background-color: #ede9e9;
      }
    }
  }
}
