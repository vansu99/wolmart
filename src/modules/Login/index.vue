<template>
  <div class="login">
    <div class="login-wrapper">
      <h2 class="login-heading">Admin</h2>
      <div class="login-error" v-if="errorMsg.length > 0">
        {{ errorMsg }}
      </div>
      <div class="login-form">
        <ValidationObserver v-slot="{ handleSubmit }">
          <form @submit.prevent="handleSubmit(onSubmit)">
            <ValidationProvider
              class="form-group"
              name="Email"
              rules="required|email"
              v-slot="{ errors }"
              tag="div"
            >
              <label class="form-label" for="email">Email</label>
              <input
                type="email"
                id="email"
                class="form-input"
                v-model="formData.email"
                placeholder="Nhập địa chỉ email của bạn"
              />
              <span class="form-error">{{ errors[0] }}</span>
            </ValidationProvider>
            <ValidationProvider
              class="form-group"
              name="password"
              rules="required|max:32|min:6"
              v-slot="{ errors }"
              tag="div"
            >
              <label class="form-label" for="email">Mật khẩu</label>
              <input
                type="password"
                id="password"
                class="form-input"
                v-model="formData.password"
                placeholder="Nhập mật khẩu của bạn"
              />
              <span class="form-error">{{ errors[0] }}</span>
            </ValidationProvider>
            <button class="form-button" type="submit">đăng nhập</button>
          </form>
        </ValidationObserver>
      </div>
    </div>
  </div>
</template>

<script>
import { userApis } from '@/apis';
import Nprogress from 'nprogress';

export default {
  name: 'Login',
  data() {
    return {
      formData: {
        email: '',
        password: '',
      },
      errorMsg: '',
    };
  },
  methods: {
    async onSubmit() {
      try {
        Nprogress.start();
        const response = await userApis.login(this.formData);
        if (response.status === 200) {
          await this.$store.dispatch('auth/login', response.data);
          await this.$router.push({ name: 'Home' });
        }
      } catch ({ error }) {
        this.errorMsg = error.message;
      } finally {
        Nprogress.done();
      }
    },
  },
};
</script>

<style lang="scss" scoped>
@import './Login';
</style>
