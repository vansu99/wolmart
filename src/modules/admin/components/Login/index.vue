<template>
  <div class="login-admin">
    <div class="login-admin-container">
      <h2 class="login-admin-title">Admin</h2>
      <div class="login-admin-form">
        <ValidationObserver v-slot="{ handleSubmit }">
          <form @submit.prevent="handleSubmit(onSubmit)" autocomplete="off">
            <ValidationProvider
              class="form-group"
              name="Email"
              rules="required|email"
              v-slot="{ errors }"
              tag="div"
              mode="passive"
            >
              <label class="form-label" for="email">Email</label>
              <input
                type="text"
                id="email"
                class="form-input"
                :class="{ error: errors[0] }"
                v-model="formData.email"
                placeholder="Nhập địa chỉ email của bạn"
              />
              <span class="form-error">{{ errors[0] }}</span>
            </ValidationProvider>
            <ValidationProvider
              class="form-group"
              name="password"
              rules="required|max:32|min:6"
              v-slot="{ errors }"
              tag="div"
              mode="passive"
            >
              <label class="form-label" for="email">Mật khẩu</label>
              <input
                id="password"
                type="password"
                v-model="formData.password"
                :class="{ error: errors[0] }"
                class="form-input input-password"
                placeholder="Nhập mật khẩu của bạn"
              />
              <span class="form-error">{{ errors[0] }}</span>
            </ValidationProvider>
            <button type="submit" class="login-admin-btn">Đăng nhập</button>
          </form>
        </ValidationObserver>
      </div>
    </div>
  </div>
</template>

<script>
import { userApis } from '@/apis';
import { setToken } from '@/utils/storage';

export default {
  name: 'LoginAdmin',
  data() {
    return {
      formData: {
        email: '',
        password: '',
      },
      errorMsg: '',
    };
  },
  methods: {
    async onSubmit() {
      try {
        const response = await userApis.login(this.formData);
        if (response.status === 200) {
          await this.$store.dispatch('auth/login', response.data);
          setToken(response.data.token);
          await this.$router.push({ name: 'AdminDashboard' });
        }
      } catch ({ error }) {
        this.errorMsg = error.message;
      }
    },
  },
};
</script>

<style lang="scss" scoped>
.login-admin {
  padding: 6rem 0;
  border-top: 1px solid rgba(0, 0, 0, 0.2);

  &-container {
    margin: 0 auto;
    max-width: 30rem;
  }

  &-form {
    .form-error {
      font-size: 1.2rem;
      font-weight: 600;
      color: #ff5252;
      display: block;
      padding-top: 0.4rem;
    }
  }

  &-title {
    font-size: 2.4rem;
    margin-bottom: 2rem;
  }

  &-btn {
    display: block;
    border: 0;
    color: #fff;
    background-color: #336699;
    width: 100%;
    padding: 1rem;
    font-size: 1.6rem;
    border-radius: 3px;
    outline: none;
  }
}
</style>
