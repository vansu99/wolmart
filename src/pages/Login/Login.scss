.login {
  position: relative;
  @include flexCenter;
  height: 100%;
  background: #8db9fc;
  background: -webkit-linear-gradient(to left, #8db9fc, #2066db);
  padding: 3rem;
  &-container {
    @include flexCenter;
    position: relative;
    max-width: 80rem;
    width: 80rem;
    margin: auto;
    border-radius: 1rem;
    box-shadow: rgba(17, 17, 26, 0.1) 0px 4px 16px, rgba(17, 17, 26, 0.1) 0px 8px 24px,
      rgba(17, 17, 26, 0.1) 0px 16px 56px;
    background: #fff;
  }
  &-wrapper {
    flex-shrink: 0;
    width: 35rem;
    padding: 3rem 3.5rem;
  }
  &-heading {
    font-size: 2.6rem;
    margin-bottom: 0.3rem;
    color: $hover-color;
    font-weight: bold;
    text-align: center;
    & img {
      width: 14.4rem;
      height: auto;
    }
  }
  &-description {
    margin-bottom: 1.5rem;
    font-family: $font-primary;
    font-size: 1.3rem;
    color: #c9c9c9;
    text-align: center;
    font-style: italic;
  }
  &-error {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
    border-radius: 4px;
    margin-bottom: 2rem;
    font-size: 1.3rem;
    padding: 10px;
  }
  &-form {
    .form-group {
      position: relative;
      margin-bottom: 2rem;
      .password-icon {
        position: absolute;
        top: 3rem;
        right: 1rem;
        color: #999;
        font-size: 1.5rem;
        cursor: pointer;
      }
    }
    .form-label {
      display: block;
      font-size: 1.3rem;
      margin-bottom: 0.2rem;
      color: $text-light;
    }
    .form-input {
      width: 100%;
      padding: 10px 10px 10px 40px;
      font-family: $font-primary;
      color: $text-extra-dark;
      outline: none;
      border: none;
      border-bottom: 1px solid #c9c9c9;
      border-radius: 0;
      overflow: hidden;
      background-color: transparent;
      transition: border-color 0.2s ease-in;
      &:focus {
        outline: none;
        background-color: transparent;
        border-color: $hover-color;
        color: $text-semi-light;
        box-shadow: $box-shadow;
      }
      &:focus + .form-icon {
        color: $text-semi-light;
      }
      &.error {
        outline: none;
        background-color: transparent;
        border-color: #a94442;
      }
      &.input-password {
        padding-right: 3.5rem;
      }
    }
    .form-icon {
      position: absolute;
      top: 3rem;
      left: 1rem;
      color: $text-light;
      font-size: 1.5rem;
    }
    .form-error {
      display: block;
      color: $bg-danger;
      font-size: 1.3rem;
      font-weight: 500;
      margin-top: 10px;
    }
    .form-forgot {
      font-size: 1.3rem;
      margin-top: -1rem;
      margin-bottom: 2rem;
      text-align: right;
      color: darkslategray;
      font-weight: normal;
      font-style: italic;
      b {
        color: $bg-primary;
        font-weight: bold;
        cursor: pointer;
      }
    }
  }
  .line-text {
    text-align: center;
    position: relative;
    padding-top: 1rem;
    span {
      display: inline-block;
      background-color: white;
      position: relative;
      z-index: 2;
      padding: 1rem;
      font-weight: 600;
      font-size: 1.5rem;
      text-transform: capitalize;
      color: #999;
    }
    &:before {
      content: '';
      width: 100%;
      height: 1px;
      position: absolute;
      top: 50%;
      left: 0;
      transform: translateY(-50%);
      background-color: #999;
    }
  }
  &-social {
    @include flexCenter;
    .social-item {
      cursor: pointer;
      @include flexCenter;
      padding: 0.6rem;
      margin-bottom: 1.2rem;
      border-radius: 50%;
      border: 1px solid transparent;
      font-size: 1.5rem;
      text-align: center;
      background-color: #fff;
      transition: border 0.2s ease-in-out;
      transition: background-color 0.2s ease-in-out;
      &:first-of-type {
        margin-right: 1rem;
      }
      &:hover {
        box-shadow: $box-shadow;
        border: 1px solid #ede9e9;
        background-color: #ede9e9;
      }
      &:hover svg {
        transition: transform 0.2s ease-in-out;
        transform: scale(1.1);
      }
    }
  }
  &-image {
    background: url(../../assets/images/Authentication/login.jpg);
    flex: 1;
    height: 75vh;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
  }
  &-register {
    font-size: 1.3rem;
    text-align: center;
    color: darkslategray;
    font-weight: normal;
    b {
      color: $bg-primary;
      font-weight: bold;
      cursor: pointer;
    }
  }
}


.circles {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.circles li {
  position: absolute;
  display: block;
  list-style: none;
  border-radius: 0.5rem;
  width: 20px;
  height: 20px;
  background: rgba(255, 255, 255, 0.2);
  transform: rotate(35deg);
}

.circles li:nth-child(1) {
  left: 25%;
  width: 80px;
  height: 80px;
  transform: translateY(10px) translateX(-300px)  rotate(20deg);
}

.circles li:nth-child(2) {
  left: 10%;
  width: 20px;
  height: 20px;
  transform: translateY(200px) translateX(-300px)  rotate(45deg);
}

.circles li:nth-child(3) {
  left: 70%;
  width: 20px;
  height: 20px;
  transform: translateY(50px) translateX(200px) rotate(25deg);
}

.circles li:nth-child(4) {
  left: 40%;
  width: 60px;
  height: 60px;
  transform: translateY(380px) translateX(-400px) rotate(15deg);
}

.circles li:nth-child(5) {
  left: 65%;
  width: 20px;
  height: 20px;
  transform: translateY(120px) translateX(350px) rotate(75deg);
}

.circles li:nth-child(6) {
  left: 75%;
  width: 110px;
  height: 110px;
  transform: translateY(200px) translateX(30px)  rotate(45deg);
}

.circles li:nth-child(7) {
  left: 35%;
  width: 150px;
  height: 150px;
  transform: translateY(400px) translateX(-500px) rotate(35deg);
}

.circles li:nth-child(8) {
  left: 50%;
  width: 25px;
  height: 25px;
  transform: translateY(350px) translateX(500px) rotate(85deg);
}

.circles li:nth-child(9) {
  left: 20%;
  width: 15px;
  height: 15px;
  transform: translateY(200px) rotate(25deg);
}

.circles li:nth-child(10) {
  left: 85%;
  width: 150px;
  height: 150px;
  transform: translateY(400px) rotate(80deg);
}

