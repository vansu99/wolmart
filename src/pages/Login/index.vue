<template>
  <div class="login">
    <div class="login-container">
      <div class="login-image">
        <img src="../../assets/images/Authentication/login.jpg" alt="bg-login" />
      </div>
      <div class="login-wrapper">
        <div class="login-heading"><img src="@/assets/images/Header/logo.png" alt="Wolmart" /></div>
        <div class="login-error" v-if="errorMsg.length > 0">
          {{ errorMsg }}
        </div>
        <div class="login-description">Người bạn của mọi nhà</div>
        <div class="login-form">
          <ValidationObserver v-slot="{ handleSubmit }">
            <form @submit.prevent="handleSubmit(onSubmit)" autocomplete="off">
              <ValidationProvider
                class="form-group"
                name="Email"
                rules="required|email"
                v-slot="{ errors }"
                tag="div"
                mode="passive"
              >
                <label class="form-label" for="email">Email</label>
                <input
                  type="text"
                  id="email"
                  class="form-input"
                  :class="{ error: errors[0] }"
                  v-model="formData.email"
                  placeholder="Nhập địa chỉ email của bạn"
                />
                <icon name="email" class="form-icon" />
                <span class="form-error">{{ errors[0] }}</span>
              </ValidationProvider>
              <ValidationProvider
                class="form-group"
                name="password"
                rules="required|max:32|min:6"
                v-slot="{ errors }"
                tag="div"
                mode="passive"
              >
                <label class="form-label" for="email">Mật khẩu</label>
                <input
                  id="password"
                  class="form-input input-password"
                  :class="{ error: errors[0] }"
                  v-model="formData.password"
                  placeholder="Nhập mật khẩu của bạn"
                  :type="hide ? 'text' : 'password'"
                />
                <icon name="pwd" class="form-icon" />
                <i class="fas fa-eye password-icon" v-if="hide" @click="hide = false"></i>
                <i class="fas fa-eye-slash password-icon" v-else @click="hide = true"></i>
                <span class="form-error">{{ errors[0] }}</span>
              </ValidationProvider>
              <p class="form-forgot">
                Quên mật khẩu? <a>Nhấn vào <b>đây</b></a>
              </p>
              <Button content="đăng nhập" />
            </form>
          </ValidationObserver>
        </div>
        <div class="line-text">
          <span>hoặc</span>
        </div>
        <div class="login-social">
          <a class="social-item">
            <icon name="google" />
          </a>
          <a class="social-item">
            <icon name="facebook" />
          </a>
        </div>
        <div class="login-register">
          <router-link to="/register">Chưa có tài khoản? <b>Đăng ký</b></router-link>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import { userApis } from '@/apis';
import Icon from '@/components/SvgIcons';
import { setToken } from "@/utils/storage";
import Button from '@/components/Button/ButtonPrimary';

export default {
  name: 'Login',
  data() {
    return {
      hide: false,
      formData: {
        email: '',
        password: '',
      },
      errorMsg: '',
    };
  },
  methods: {
    async onSubmit() {
      try {
        const response = await userApis.login(this.formData);
        if (response.status === 200) {
          await this.$store.dispatch('auth/login', response.data);
          setToken(response.data.token);
          await this.$router.push({ name: 'Home' });
        }
      } catch ({ error }) {
        this.errorMsg = error.message;
      }
    },
  },
  components: { Icon, Button },
};
</script>

<style lang="scss" scoped>
@import './Login';
</style>
