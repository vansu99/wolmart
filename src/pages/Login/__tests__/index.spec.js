import Login from '@/pages/Login/index';
import flushPromises from 'flush-promises';
import { shallowMount } from '@vue/test-utils';
import wrapperOps from '../../../../test/wrapper';
import Button from '@/components/Button/ButtonPrimary';
import { ValidationProvider, ValidationObserver } from 'vee-validate';

// const userMock = {
//   id: 1,
//   name: 'evan',
//   email: 'evan@gmail.com',
//   token: '2lsaW5nLWJyb29rLTg4Mzg2Lmhlcm9rdWFwcC5jb21cL2FwaVwvbG9naW4iLCJpYXQiOjE2ND',
// };

// jest.mock('axios', () => ({
//   __esModule: true,
//   post: jest.fn(() => userMock),
// }));

describe('Login.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Login, {
      ...wrapperOps,
      stubs: {
        Button,
        ValidationObserver,
        ValidationProvider,
      },
    });
  });

  afterEach(() => {
    wrapper.vm.$destroy();
  });

  describe('#mounted', () => {
    it('should render component', () => {
      expect(wrapper.exists()).toBe(true);
      expect(wrapper.vm).toBeTruthy();

      // check components

      // check snapshot
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('#check validate for fields', () => {
    it('should show error when email is empty', async () => {
      await wrapper.find('#email').setValue('');
      await wrapper.find('#password').setValue('123456');

      const { formData } = wrapper.vm;
      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.find('.form-error').text();
      expect(formData.email).toEqual('');
      expect(formData.password).toEqual('123456');
      expect(errorMessage).toContain('The Email field is required');
    });

    it('should show error when email is invalid', async () => {
      await wrapper.find('#email').setValue('test');
      await wrapper.find('#password').setValue('123456789');

      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.find('.form-error').text();
      expect(errorMessage).toContain('The Email field must be a valid email');
    });

    it('should not error when email is valid', async () => {
      await wrapper.find('#email').setValue('test@gmail.com');
      await wrapper.find('#password').setValue('1234567');

      const { formData } = wrapper.vm;
      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.find('.form-error').text();

      expect(formData.email).toEqual('test@gmail.com');
      expect(errorMessage).toContain('');
    });

    it('should show error when password is empty', async () => {
      await wrapper.find('#email').setValue('test@gmail.com');
      await wrapper.find('#password').setValue('');

      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.vm.$refs.provider.errors[0];
      expect(errorMessage).toContain('The Password field is required');
    });

    it('should show error when password invalid', async () => {
      await wrapper.find('#email').setValue('test@gmail.com');
      await wrapper.find('#password').setValue('1234');

      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.vm.$refs.provider.errors[0];
      expect(errorMessage).toContain('The Password field must be at least 6 characters');
    });

    it('should show error when password greater than 32 characters', async () => {
      await wrapper.find('#email').setValue('test@gmail.com');
      await wrapper
        .find('#password')
        .setValue('123456789123456789123456789123456789123456789123456789');

      await wrapper.find('form').trigger('submit.prevent');
      await flushPromises();

      const errorMessage = wrapper.vm.$refs.provider.errors[0];
      expect(errorMessage).toContain(
        'The Password field may not be greater than 32 characters'
      );
    });
  });

  describe('#click button submit', () => {
    // it('should call submit method', async () => {
    //   await wrapper.find('form').trigger('submit.prevent');
    //   await flushPromises();
    //
    //   mockAxios.onPost('/login').reply(200, userMock)
    //   const result = await userApis.login(userMock)
    //   console.log(result)
    // });
  });
});
