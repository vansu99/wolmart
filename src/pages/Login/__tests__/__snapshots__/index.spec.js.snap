// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue #mounted should render component 1`] = `
<div class="login">
  <div class="login-container">
    <div class="login-image"><img src="../../assets/images/Authentication/login.jpg" alt="bg-login"></div>
    <div class="login-wrapper">
      <div class="login-heading"><img src="@/assets/images/Header/logo.png" alt="Wolmart"></div>
      <!---->
      <div class="login-description">Người bạn của mọi nhà</div>
      <div class="login-form"><span><form autocomplete="off"><div class="form-group"><label for="email" class="form-label">Email</label> <input type="text" id="email" placeholder="Nhập địa chỉ email của bạn" class="form-input"> <icon-stub name="email" width="25" height="25" class="form-icon"></icon-stub> <span class="form-error"></span></div>
      <div class="form-group"><label for="password" class="form-label">Mật khẩu</label> <input id="password" placeholder="Nhập mật khẩu của bạn" type="password" class="form-input input-password">
        <icon-stub name="pwd" width="25" height="25" class="form-icon"></icon-stub> <i class="fas fa-eye-slash password-icon"></i> <span class="form-error"></span>
      </div>
      <p class="form-forgot">
        Quên mật khẩu? <a>Nhấn vào <b>đây</b></a></p> <button type="submit" class="btn full medium">
        đăng nhập
      </button></form></span>
    </div>
    <div class="line-text"><span>hoặc</span></div>
    <div class="login-social"><a class="social-item">
        <icon-stub name="google" width="25" height="25"></icon-stub>
      </a> <a class="social-item">
        <icon-stub name="facebook" width="25" height="25"></icon-stub>
      </a></div>
    <div class="login-register">
      <router-link-stub to="/register" tag="a" ariacurrentvalue="page" event="click">Chưa có tài khoản? <b>Đăng ký</b></router-link-stub>
    </div>
  </div>
</div>
</div>
`;
